### PR TITLE
Improve exec on JRuby

### PR DESCRIPTION
**Pull request by ojab**
Unfortunately I can't get testsuite running locally, hope that travis will tell that changes are OK.
Fixes https://github.com/bundler/bundler/issues/4586.

I don't really know if test should be added for the last commit, please advise.

### CODE REVIEW

**Comment by ojab** [comment]
Ouch, JRuby is not tested on travis :/.
Anyway, manual testing of exec on JRuby works fine locally and changes are not too invasive, so hope that's ok.

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -85,10 +85,6 @@
   original_wd  = Dir.pwd
   original_env = ENV.to_hash
 
-  def pending_jruby_shebang_fix
```
Comment (segiddins) [inline spec/spec_helper.rb]: Why did you remove this?

Comment (ojab) [inline spec/spec_helper.rb]: Oops, doesn't commited spec/support/builders.rb changes, pushed fixed version.
I assume that `executables` is actually ruby scripts here regardless of RUBY_PLATFORM, so why not check `#/usr/bin/env jruby\n` shebang on JRuby.
These test are passing here w/ `ruby_shebang?` changes.

File: lib/bundler/cli/exec.rb [hunk 2]
```diff
@@ -25,7 +25,11 @@ def run
       if bin_path = Bundler.which(cmd)
         return kernel_load(bin_path, *args) if ruby_shebang?(bin_path)
         # First, try to exec directly to something in PATH
-        kernel_exec([bin_path, cmd], *args)
+        if RUBY_PLATFORM == "java" && RUBY_VERSION < "1.9"
```
Comment (segiddins) [inline lib/bundler/cli/exec.rb]: `if Bundler.current_ruby.jruby_18?`

File: spec/support/builders.rb [hunk 3]
```diff
@@ -534,8 +534,13 @@ def executables=(val)
         @spec.executables = Array(val)
         @spec.executables.each do |file|
           executable = "#{@spec.bindir}/#{file}"
+          shebang = if RUBY_PLATFORM == "java"
```
Comment (segiddins) [inline spec/support/builders.rb]: `Bundler.current_ruby.jruby?`

**Comment by segiddins** [comment]
@homu r+
